_posts/2021-04-28-#17331.md

```diff
@@ -0,0 +1,145 @@
+      and economic considerations are opposed. If you want to understand privacy
+      considerations better, check out the [Privacy
+      article](https://en.bitcoin.it/wiki/Privacy) on the Bitcoin wiki.
+
+    - Reliably move the payment(s) to finalization:  
+      Using unconfirmed inputs can make transactions unreliable. When a foreign
+      input is used, the parent transaction may time out or get replaced. Even
+      using self-sent unconfirmed funds may delay the new transaction if the
+      parent transaction has an extensive ancestry, is extraordinarily large, or
+      used a lower feerate than targeted for the child transaction.
+
+- The process of creating a transaction in Bitcoin Core's wallet roughly
+  follows these steps:
+
+    1. `CreateTransaction()` builds the header and recipient outputs, i.e. the
+       parts of the transaction that will remain fixed throughout the coin
+       selection process.
+
+    2. The feerates are established using `GetMinimumFeeRate()` which falls back
+       either to fee estimation or user input. The feerates remain constant for
+       the remaining process. The amount to be raised from the inputs,
+       `nTargetValue` is calculated from the sum of recipient amounts, the
+       target feerate, and the size of the fixed transaction parts.
+
+    3. The wallet's UTXO pool is retrieved via `AvailableCoins()`.
+
+    4. Using the above fixed values, `SelectCoins()` is used to pick an input
+       set to fund the transaction. If the user manually selected some UTXOs by
+       means of the `CoinControl` feature, these UTXOs are used first. The
+       remaining UTXOs are grouped into `OutputGroup`s. The grouped UTXOs are
+       then used in one or multiple rounds of `SelectCoinsMinConf()` which
+       use increasingly permissive `CoinEligibilityFilter`s to preselect
+       `OutputGroup`s until an input set is found. As described above,
+       `SelectCoinsMinConf()` tries BnB first, then falls back to knapsack if no
+       solution was found.
+    5. Determine whether to create a change output, finish assembling the
+       transaction, and sign the transaction.
+
+- When a UTXO is selected as an input, the size of the transaction is increased
+  which in turn increases the necessary fees to maintain the targeted feerate.
+  We can preempt the cost added by the input by considering each UTXO at its
+  _effective value_, its value reduced by the UTXO's input cost at the given
+  feerate.
+
+- Using _effective values_ in coin selection means that we can freely add and
+  remove inputs until we have have raised enough funds to pay for the
+  transaction's fixed costs: the payment amounts, the transaction header and the
+  output data. Whatever final input set we settle for, the increased transaction
+  size and input costs have already been accounted for.
+
+- The _knapsack solver_ sorts the available UTXOs by effective value, and then
```

Comment:
Ordering: I think these could be moved up, i.e.
```
- There are 2 types of solvers
    - knapsack
    - branch and bound
- When a UTXO is selected... effective values!

```

Reply:
Reordered.